### PR TITLE
networkd: Geneve add DF feature and allow TTL to bechosen by kernel

### PR DESCRIPTION
**Pull request by ssahani**
1. Add support to configure  IP don't fragment.
2. Allow TTL to be chosen by kernel.

### CODE REVIEW

File: man/systemd.netdev.xml [hunk 1]
```diff
@@ -731,7 +731,8 @@
       <varlistentry>
         <term><varname>TTL=</varname></term>
         <listitem>
-          <para>Specifies the TTL value to use in outgoing packets. Ranges [1-255].</para>
+          <para>Specifies the TTL value to use in outgoing packets. Takes a number in the range 0-255.
+           When unset, the kernel's default will be used. </para>
```
Comment (yuwata) [inline man/systemd.netdev.xml]: `0` means unset, so I think it is not necessary to include 0 in the range. That is, I prefer `1-255`.

File: src/network/netdev/geneve.h [hunk 2]
```diff
@@ -21,16 +21,17 @@ typedef enum GeneveDF {
 struct Geneve {
         NetDev meta;
 
+        unsigned ttl;
```
Comment (yuwata) [inline src/network/netdev/geneve.h]: Why do you change the type? I think `uint8_t` is better.

**Comment by yuwata** [comment]
Maybe the explanations of `TTL=` in [Tunnel] and [VXLAN] also need to be updated.

File: man/systemd.netdev.xml (hunk 1 above)
Comment (ssahani) [reply man/systemd.netdev.xml]: Changed a bit but updated what exactly means by 0 or unset.

File: src/network/netdev/geneve.h (hunk 2 above)
Comment (ssahani) [reply src/network/netdev/geneve.h]: I wanted to print the error in the verify but that's ok

**Comment by ssahani** [comment]
yes will update. Thanks for the review @yuwata .

**Comment by yuwata** [comment]
@ssahani Slightly update the man page.
- mention that `Id=` is mandatory.
- include `0` in the range again, as you added the explanation what `0` means.

**Pull request merged**